lib/exercises.rb

```diff
@@ -1,19 +1,57 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: a lot
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash = {}
+
+  strings.each do | string | 
+
+   a_key = string.chars.sort
+
+   if hash.key? (a_key)
+     hash[a_key] << string 
+   else
+     hash[a_key] = [string]
+   end
+  end 
+
+   arrayArrays = []
+   hash.each do | key , value|
+     arrayArrays << hash[key]
+   end
+
+   return arrayArrays 
```

Comment:
Since values has the arrays of anagrams in them...

```suggestion
   return hash.values
```